### ISSUE TITLE
Create feature-downloads component

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#459]
#424 landed the code to expose downloads in `engine` and `session`.

For all our apps that want to use Android's download manager for downloads the code will be very similar and we can wrap that in a feature component.

The component will roughly do:
* Subscribe to downloads in the currently selected session (I assume we do not need to subscribe to it in *all* Sessions?)
* Consume `Download` objects and pass the data to Androids Download Manager
* Show an optional, customizable UI piece before starting the download (e.g. Download dialog)
* Call back into the app to notify if a download is done (to show some UI)
* Call back into the app to request the Storage permission if needed
* Provide a callback to be called by the app once a permission has been granted (and then it will trigger the download)

**Comment by ekager** [comment]
Came here to file this! You're always one step ahead 👍

**Comment by Amejia481** [comment]
This may give us some ideas!
https://blog.novoda.com/download-manager-v2/

**Comment by pocmo** [comment]
> This may give us some ideas!
> https://blog.novoda.com/download-manager-v2/

This may be a separate component (or a customization to be added at some later point) though! :)

**Comment by Amejia481** [comment]
Exactly, this blog post caught my attention because from it we can get some ideas from their learned experiences of creating a similar library.

**Comment by pocmo** [comment]
Closing this in favor of the two child issues. No need to keep the meta issue around.
